File: 565.数组嵌套.py
#
# @lc app=leetcode.cn id=565 lang=python3
#
# [565] 数组嵌套
#

# @lc code=start
from typing import List


class Solution:
    def arrayNesting(self, nums: List[int]) -> int:
        ans, n = 0, len(nums)
        for i in range(n):
            cnt = 0
            while nums[i] != -1:
                num = nums[i]
                cnt += 1
                nums[i] = -1
                i = num
            if ans < cnt:
                ans = cnt
        return ans
# @lc code=end

File: 565.数组嵌套.swift
/*
 * @lc app=leetcode.cn id=565 lang=swift
 *
 * [565] 数组嵌套
 */

// @lc code=start
class Solution {
    func arrayNesting(_ nums: [Int]) -> Int {
        let INF = 0xf3f3f3f3
        var ans = 0
        var arr = nums
        for i in 0..<arr.count {
            if arr[i] != INF {
                var start = arr[i]
                var count = 0
                while arr[start] != INF {
                    let tmp = start
                    start = arr[start]
                    count += 1
                    arr[tmp] = INF
                }
                ans = max(ans, count)
            }
        }
        return ans
    }
}
// @lc code=end

